packages/kit/src/utils/url.js

```diff
@@ -103,17 +103,50 @@ const tracked_url_properties = /** @type {const} */ ([
 	'href',
 	'pathname',
 	'search',
-	'searchParams',
 	'toString',
 	'toJSON'
 ]);
 
+const tracked_search_params_properties = /** @type {const} */ ['get', 'getAll', 'has'];
+
+/**
+ * @param {URLSearchParams} search_params
+ * @param {() => void} callback
+ * @param {(search_param: string) => void} search_params_callback
+ */
+function tracked_search_params(search_params, callback, search_params_callback) {
+	return new Proxy(search_params, {
+		get(obj, key) {
+			if (typeof key === 'string' && tracked_search_params_properties.includes(key)) {
+				// @ts-expect-error
+				const function_to_call = key in search_params ? search_params[key].bind(search_params) : '';
+				if (function_to_call && function_to_call instanceof Function) {
```

Comment:
I don't think we need to do these checks. This would always be true because we know all the tracked properties are functions.

Reply:
It was conservative to make sure that if in the future someone adds another non existent or non function property it doesn't break...but I can remove it :)

Reply:
Sooo...should i remove this checks?

Reply:
Yes better to remove it